### PR TITLE
Remove `#undef Success` in pcl_macros.h by extracting `PCL_MAKE_ALIGNED_OPERATOR_NEW` into eigen_base.h

### PR DESCRIPTION
**Pull request by aPonza**
Solves an issue in #3545

### CODE REVIEW

**Comment by kunaltyagi** [comment]
PTAL at my change https://github.com/PointCloudLibrary/pcl/pull/3545/commits/a4e46e7f0a4db09b1df687ae8c2d2b3c92723005 where I've moved around a different (and smaller) set of includes.

**Comment by aPonza** [comment]
@kunaltyagi so you're saying std -> library -> 3rd party instead of library -> std -> 3rd party? I remember arguments in favor of both, and I see different approaches in PCL's files' convention (if there is one).

**Comment by taketwo** [comment]
I had the impression that we want to have "library -> 3rd party -> std", but this was not codified yet.

**Comment by kunaltyagi** [comment]
The order is not the concern. The files used is one. I move Eigen/Core and you move a lot more (indirectly included via pcl_macro).

As for the order, both have pro and cons. I don't really mind changing the order. The reason I prefer std first is because I trust the files on top to not include/declare anything that might cause a problem for later includes. So my personal order is std->3rd_party->library in a decreasing order of trust.

**Comment by SergioRAgostinho** [comment]
> I had the impression that we want to have "library -> 3rd party -> std", but this was not codified yet.

This. To catch issues with potential transitive dependencies early on.

**Comment by aPonza** [comment]
I tried, it works here, let me see what CI says.

**Comment by aPonza** [comment]
Formatting is locally fixed (includes in alphabetic order), tutorials are locally fixed (forgot to add the new header in CMakeLists.txt), but Build.Ubuntu-16.04 fails for a PCL_EXPORTS error:
```
[ 16%] Building CXX object simulation/CMakeFiles/pcl_simulation.dir/src/camera.cpp.o
In file included from /__w/1/s/simulation/src/camera.cpp:2:0:
/__w/1/s/simulation/include/pcl/simulation/camera.h:12:19: error: variable 'pcl::simulation::PCL_EXPORTS pcl::simulation::Camera' has initializer but incomplete type
 class PCL_EXPORTS Camera {
                   ^
```
which I don't understand. The error is for forward declarations which require full types, but pcl_exports.h is about NVCC. I'm thinking eigen_base.h is lacking a way to avoid NVCC like pcl_macros.h used to due to Eigen/Core, but don't know what or where yet.

**Comment by kunaltyagi** [comment]
We can remove pcl_exports now that pcl_macros is about to be nvcc compatible (post deprecated PR merge #3634) with the removal of Eigen/Core.

As for the error, it seems like PCL_EXPORTS is not found (since the compiler assumes it is in simulation namespace). You might need to include pcl_macros wherever the relevant macros are used.

**Comment by aPonza** [comment]
I can build pcl_simulation, there was also a missing couple of `boost::` on top of the missing header. Should removing `PCL_EXPORTS` and header be another PR or this one?

File: simulation/include/pcl/simulation/camera.h [hunk 1]
```diff
@@ -5,14 +5,15 @@
...
+#include <pcl/pcl_exports.h>
 
 namespace pcl {
 namespace simulation {
 
 class PCL_EXPORTS Camera {
 public:
-  using Ptr = shared_ptr<Camera>;
-  using ConstPtr = shared_ptr<const Camera>;
+  using Ptr = boost::shared_ptr<Camera>;
```
Comment (taketwo) [inline simulation/include/pcl/simulation/camera.h]: It was intentional that `shared_ptr` is used without namespace. Before your previous commit, this file included "pcl_macros.h" header which provided `pcl::shared_ptr`.

File: common/include/pcl/eigen_base.h [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  inline pcl::Vector3fMap getNormalVector3fMap () { return (pcl::Vector3fMap (data_n)); } \
+  inline pcl::Vector3fMapConst getNormalVector3fMap () const { return (pcl::Vector3fMapConst (data_n)); } \
+  inline pcl::Vector4fMap getNormalVector4fMap () { return (pcl::Vector4fMap (data_n)); } \
+  inline pcl::Vector4fMapConst getNormalVector4fMap () const { return (pcl::Vector4fMapConst (data_n)); }
+
+#define PCL_ADD_NORMAL4D \
+  PCL_ADD_UNION_NORMAL4D \
+  PCL_ADD_EIGEN_MAPS_NORMAL4D
+
+#define PCL_ADD_UNION_RGB \
```
Comment (taketwo) [inline common/include/pcl/eigen_base.h]: This macro has nothing to do with Eigen. I think only the macros that have `EIGEN_MAPS` in the name should be here.

File: common/include/pcl/eigen_base.h [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+ *
+ * \see pcl::has_custom_allocator, pcl::make_shared
+ * \ingroup common
+ */
+#define PCL_MAKE_ALIGNED_OPERATOR_NEW \
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW \
+  using _custom_allocator_type_trait = void;
+
+#define PCL_ADD_EIGEN_MAPS_POINT4D \
+  inline pcl::Vector3fMap getVector3fMap () { return (pcl::Vector3fMap (data)); } \
```
Comment (taketwo) [inline common/include/pcl/eigen_base.h]: This relies on `Vector3fMap` that is declared in "point_types.hpp". So either include that file or, move the `using`s here. The latter makes more sense to me.

**Comment by kunaltyagi** [comment]
> What do you think about eigen.h instead of eigen_base.h

Not opposed. But it'd require renaming the existing `eigen.h` to something else like `eigen_operations.h` or similar

> I'm not sure if global replace of pcl_macros.h with eigen_base.h is reasonable.

It's not. At the very minimum all files without pcl_macros but with eigen_base and PCL_EXPORT will need pcl_macros.

**Comment by taketwo** [comment]
I have a feeling we are going to make some hastily moves now for no good reason and break something. Let's limit this PR to removing undefs and fixing the order of includes in VTK header. Reorganization of `pcl_macros.h`, `pcl_exports.h`, `point_traits.h`, etc is a bigger topic that has to be handled with care.

File: simulation/include/pcl/simulation/camera.h (hunk 1 above)
Comment (aPonza) [reply simulation/include/pcl/simulation/camera.h]: One point in favour of an eventual memory.h made out of shared_ptr.h and make_shared.h.

File: common/include/pcl/eigen_base.h (hunk 2 above)
Comment (aPonza) [reply common/include/pcl/eigen_base.h]: Ok, that was why my comment was asking for the line range.

**Comment by aPonza** [comment]
> What do you think about `eigen.h` instead of `eigen_base.h`?

As Kunal points out (and I did too, [here](https://github.com/PointCloudLibrary/pcl/pull/3545#issuecomment-586949432)) there are already files named eigen.h in io, common, surface, features, geometry, registration, visualization, sample consensus and in (one of the) apps.

> Let's limit this PR

Eh, agreed, seems to me like ~130 files are a tad too many to check by hand. Unless IWYU can handle it all? Then I add the new header instead of replacing pcl_macros, so it's at most a useless include which was already present and IWYU would/should remove it with a pass. I can commit that and maybe Kunal could try a pass on the PR with it to check?

**Comment by aPonza** [comment]
[The errors](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=11515&view=logs&j=4de25620-fe7f-583f-ecca-01dec01cc67e&t=6d22f93b-8a70-51cf-73f0-01dd97334d5e&l=3374) don't seem related to my PR:
```
test 84
        Start  84: outofcore_test

84: Test command: /__w/1/build/test/outofcore/test_outofcore
84: Test timeout computed to be: 10000000
84: [==========] Running 16 tests from 2 test cases.
84: [----------] Global test environment set-up.
84: [----------] 4 tests from PCL

  [...]

84: [----------] 4 tests from PCL (837 ms total)
84: 
84: [----------] 12 tests from OutofcoreTest
84: [ RUN      ] OutofcoreTest.Outofcore_Constructors
84: unknown file: Failure
84: C++ exception with description "boost::filesystem::create_directory: No space left on device: "treeA/7/6/7/7/1/1/7/3/7/2"" thrown in the test body.
84: [  FAILED  ] OutofcoreTest.Outofcore_Constructors (14162 ms)
84: [ RUN      ] OutofcoreTest.Outofcore_ConstructorSafety
84: /__w/1/s/test/outofcore/test_outofcore.cpp:482: Failure
84: Value of: boost::filesystem::exists (filename_otreeB)
84:   Actual: false
84: Expected: true
84: No tree detected on disk. This test will fail. Perhaps this test was run out of order.
84: 
84: [  FAILED  ] OutofcoreTest.Outofcore_ConstructorSafety (0 ms)
84: [ RUN      ] OutofcoreTest.Outofcore_ConstructorBadPaths
84: [pcl::outofcore::OutofcoreOctreeBaseNode] Could not find dir treeBogus
84: [pcl::outofcore::OutofcoreOctreeBase] Wrong root node file extension: .bad_extension. The tree must have a root node ending in .oct_idx
84: [       OK ] OutofcoreTest.Outofcore_ConstructorBadPaths (0 ms)

  [...]

84: [----------] 12 tests from OutofcoreTest (16765 ms total)
84: 
84: [----------] Global test environment tear-down
84: [==========] 16 tests from 2 test cases ran. (17602 ms total)
84: [  PASSED  ] 14 tests.
84: [  FAILED  ] 2 tests, listed below:
84: [  FAILED  ] OutofcoreTest.Outofcore_Constructors
84: [  FAILED  ] OutofcoreTest.Outofcore_ConstructorSafety
84: 
84:  2 FAILED TESTS
 84/115 Test  #84: outofcore_test .........................***Failed   18.05 sec
```

**Comment by kunaltyagi** [comment]
> but my text editor strips extra spaces.

Mine too (Don't know if I should cry or be happy about this)

**Comment by taketwo** [comment]
OK, the taken "global replace" approach seems solid to me and I'm not afraid that something is going to break.

However, and please forgive me for being stubborn, I do not understand the motivation behind:
  * extracting this single macro into a file of it's own (`PCL_MAKE_ALIGNED_OPERATOR_NEW` **is** a macro, so logically belongs to `pcl_macros.h`  I think);
  * name choice for this header file (which meaning does "base" carry in this name?)

**Comment by kunaltyagi** [comment]
IMO, pcl_macros is a bad example of a filename. There are several kinds of macros and each brings with it a different header baggage. There are mainly 4 kinds of macros in PCL
* Compiler support macros: PCL_DEPRECATED("This stuff is bad for you"), PCL_FALLTHROUGH, PCL_EXPORTS etc.
* Library support macros: PCL_MAKE_EIGEN_TYPES_WORK (the macro in question), and others
* Quality of code support macros: PCL_ADD_NORMAL_4D, etc.
* Tooling macros: provided by CMake (in pcl_config.h), built on spot (for OpenMP, SSE, etc.)

Clearly, the last 2 types have been segregated by origin, use and functionality. The filename isn't a good argument for putting all macros everywhere in one file. Macros are just like functions. There's no harm in creating one file for a function if it doesn't fit in with 10 functions with a similar input arguments but different functionality.

**Comment by aPonza** [comment]
>     extracting this single macro into a file of its own

While there would be no harm in doing this per se (some go so far as to advocate for one function per file to avoid including unnecessary headers, even...a bit too much for me), my intention was to predispose a file to separate this macro together with the ones in point_types.hpp which were previously included. Doing so in two separate PRs allows a more granular check for breakage (as you say yourself, this micro-move I've made made you "not afraid that something is going to break", which was a concern for me as well with how the PR was set up 2 days ago).

If you think the PCL_MAKE_EIGEN_TYPES_WORK macros should be bundled together (e.g. to include Eigen/Core only there, but also to group logically similar macros together) then it's just a matter of choosing a fitting name for the header, which brings me to

>     which meaning does "base" carry in this name?

None really, maybe eigen_support.h, eigen_types.h (we use the macro to define an allocator trait, and the eigen vector maps are types too), or anything else that works better. My point in pushing with eigen_base.h was to at least confirm nothing breaks and the methodology is sound. It doesn't take much to change the name once one is decided. This PR is still a draft, if you will.

Moreover anybody that uses IWYU can already check the modified headers, where they might find some `#include <pcl/pcl_macros.h>` can be removed because the prior inclusion was only needed for `PCL_MAKE_ALIGNED_OPERATOR_NEW`.

**Comment by taketwo** [comment]
This PR started like a simple fix (move the include up, remove undefs) and in a blink of an eye spiraled into a massive changeset with over 100 files touched. On top of that, there is a parallel PR by Kunal. I was overwhelmed.

Now with your clarifications and extended opinions, I have a better understanding of your ideas, thank you. I do agree that there are different kinds of macros and that there is logic in being more granular and organizing them in separate files. I would support a principled reorganization of the headers. But I feel like we are missing some rough plan of where we are going.

**Comment by kunaltyagi** [comment]
> On top of that, there is a parallel PR by Kunal

Sorry about my PRs. I've been moving slowly on them because they aren't automatic fixes and finicky compilation issues (see the 3 week old PR on grabber) really hamper productivity. The PR referred to consists of reorganization of the includes, not the headers and not other modules (beyond making things compile). As such, I did the bare minimum and would prefer another PR to attack the issue better.

> I would support a principled reorganization of the headers. But I feel like we are missing some rough plan of where we are going.

One problem that I just realized is that PCL doesn't define which macros are public and which are private. That'd be the first step in making a transition plan.

To clarify my thoughts about this PR

* target: remove Eigen header from pcl_macros
* Why? pcl_macros is getting large and affecting multiple PR, implying it needs refactoring
* process:
  * move Eigen based macros to a new file: name? location? private/public?
  * update other files to include the new header (100+ files need updates)
* bonus:
  * move all macros that might be used with eigen together (not needed, but will help in making decisions both now and later (like filename, location, etc.))
  * remove pcl_exports and update includes (not needed here since I have done that in a separate commit and can create a PR of it once this PR is accepted)
* related PR:
  * adding `index_t` since adding new type to pcl_macros doesn't make sense (I added one type shared_ptr, but I now know that was an error)
  * update `#include`s using IWYU since it keeps getting confused between the 2

The other 2 PR will be mostly merged after a decision has been made on this one since
* they still need work (compilation, discussion, etc.)
* this PR is less broad (no larger reorganization, 'only' 800 lines of diff) as of now

As such any changes made in the parallel PR wrt pcl_macros can be ignored as I'm making them to keep the direction of my PRs consistent (not necessarily with this one).

**Comment by taketwo** [comment]
> target: remove Eigen header from pcl_macros

:+1: 

> * move Eigen based macros to a new file: name? location? private/public?
> * move all macros that might be used with eigen together (not needed, but will help in making decisions both now and later (like filename, location, etc.))

I guess have a different view on Eigen-related macros. We don't have many of them. There is `PCL_MAKE_ALIGNED_OPERATOR_NEW` in "pcl_macros.h" and there is a family of `PCL_ADD_EIGEN_MAPS_xxx` in "point_types.hpp"

* The former is Eigen-related, yes. However, it's also `make_shared`-related. In fact, the users who don't care about using `pcl::make_shared` don't have any reason to use this macro instead of the standard one. Therefore, I think it should instead go together with `shared_ptr`, `make_shared`, etc. Perhaps STL-like "memory.h" is an appropriate name choice.
* I don't see much benefit from moving the latter. These macros are helpers for creation/registration of PCL point types and are only used in "point_types.hpp". They are not exactly "private" because potentially the user may create his own point type and add these maps. However, the user will use them together with other stuff from "point_types.hpp", so it's not like he will be granular and include "eigen_base.h", but not "point_types.hpp".

**Comment by aPonza** [comment]
Are you thinking something like memory.h ([`pcl::make_shared`](https://github.com/PointCloudLibrary/pcl/blob/b0fe402055fa3e9cbd91746f19a3d0bf4d465883/common/include/pcl/make_shared.h), [`using shared_ptr`](https://github.com/PointCloudLibrary/pcl/blob/b0fe402055fa3e9cbd91746f19a3d0bf4d465883/common/include/pcl/pcl_macros.h#L81), the pcl wrapped eigen macro, ...) and type_traits.h (which is **not** a concern at the moment apart from the trait being related to this, which includes [`pcl::has_custom_allocator`](https://github.com/PointCloudLibrary/pcl/blob/1e3c62e2df0c739a7f5fd13610b043408d03aebe/common/include/pcl/point_traits.h#L343) (at the moment it's in point_traits.h but there was agreement in it not belonging there fully ([#3163 comment](https://github.com/PointCloudLibrary/pcl/pull/3163#issuecomment-502065051)), ...)?

Not sure if there would be value in separating inside memory.h or if everything should just be dumped in. Ping for @SergioRAgostinho who had a strong opinion on separating `make_shared` from the eigen allocation macro. Separating them would also allow to not touch as many files (i.e. those that only use `pcl::make_shared` can continue doing so in their including make_shared.h).

From a user standpoint I'd use `pcl::make_shared` regardless of the rest, and `std::`/`boost::` pointers (which are likely already in my user code): I'm not interested in the library's implementation details for custom allocation, therefore I'd still only want to only `#include <pcl/make_shared.h>`. From a library developer I'd instead want to be able to include all the things since when I'm writing a class with alignment concerns, I'd want all three (four via dependent include) things.

**Comment by SergioRAgostinho** [comment]
> Ping for @SergioRAgostinho who had a strong opinion on separating `make_shared` from the eigen allocation macro. Separating them would also allow to not touch as many files (i.e. those that only use `pcl::make_shared` can continue doing so in their including make_shared.h).

Because that separation allows using  an "allocation aware" `make_shared` without ever being exposed to Eigen definitions. There will still be cases where `make_shared` is used with types which are not in any way related to Eigen and in those situations it would be great to not expose Eigen types if it is not required.

Regarding the final location of the `PCL_MAKE_ALIGNED_OPERATOR_NEW`, if a class has eigen members, it's likely that it will need to define `EIGEN_MAKE_ALIGNED_OPERATOR_NEW`. So whenever Eigen objects exist, it makes sense for the `PCL_MAKE_ALIGNED_OPERATOR_NEW` to also exists. If this is the case, for me it makes sense to have this specific macro in whatever eigen core header comes out of this PR.

As a final note, I would keep traits in their own isolated header as much as possible. Definitely not in memory.h

**Comment by kunaltyagi** [comment]
If and when we target C++17, then we don't need the `pcl_make_aligned_operator_new`. See [eigen macro](https://bitbucket.org/eigen/eigen/src/94875feeeeb9abe5509b314197da1991ba2070f5/Eigen/src/Core/util/Memory.h#lines-771)

> keep traits in their own isolated header as much as possible. Definitely not in memory.h

👍

**Comment by aPonza** [comment]
> If and when we target C++17, then we don't need the `pcl_make_aligned_operator_new`

[#3108-comment](https://github.com/PointCloudLibrary/pcl/pull/3108#issuecomment-496866990) :) It will still probably be needed for `#if (__cplusplus < 201703L)` but yes, it's kind of for naught.

**Comment by taketwo** [comment]
> Are you thinking something like memory.h ... and type_traits.h  ... which includes pcl::has_custom_allocator

No, I'd put the custom allocator trait into "memory.h" as well. Why? This is a bit of a false dichotomy, "memory.h" vs "type_traits.h". The former represents topic-based grouping of entities (in this case: memory allocation and management), whereas the latter is type-based grouping (in this case: structs that serve as traits in meta-programming). Just a few days ago we've agreed that type-based grouping is no good. That's why "pcl_macros.h" has to be abolished.  Therefore, I don't feel like supporting existence of "type_traits.h" by moving there something that would naturally fit into topic-based "memory.h".

> Because that separation allows using an "allocation aware" make_shared without ever being exposed to Eigen definitions. There will still be cases where make_shared is used with types which are not in any way related to Eigen and in those situations it would be great to not expose Eigen types if it is not required.

Eigen is at the core of PCL. I challenge you to explain how I can use any bit of functionality provided by PCL without including Eigen :laughing:. So the user of PCL is exposed to Eigen, that's the reality. I don't think someone will depend on PCL to make use of `pcl::make_shared`, but not any other PCL type.

**Comment by kunaltyagi** [comment]
If I'm not wrong, C++ STL achieve this effect by including smaller headers in larger ones. There will be a `memory_type_traits.h` which is included both by memory and type_traits to reduce compilation load. But for the sake of public API, it is exposed only in one, and the other is "hidden", eg: behind a namespace. This isn't really true of all API points (some are exposed in several headers), but mostly yes.

For the most over-the-top example, see the user-defined-literals.

**Comment by aPonza** [comment]
Do I see an agreement? :) Should I go ahead and put everything inside memory.h but with separate headers like the STL? I would leave make_shared.h (`pcl::make_shared`), shared_ptr.h (`using` declaration), and memory_type_traits.h (`pcl::has_custom_allocator`) but there's a missing name, of course, for the wrapped macro. Should it be left inside memory.h directly maybe?

**Comment by SergioRAgostinho** [comment]
I'm not sure about the separate micro headers at this early stage but I'm ok in proceeding with it bundling everything into `memory.h` if everyone else is ok with it.

**Comment by aPonza** [comment]
```
In file included from /opt/libpcl/pcl/octree/include/pcl/octree/octree2buf_base.h:43:0,
                 from /opt/libpcl/pcl/octree/include/pcl/octree/octree.h:41,
                 from /opt/libpcl/pcl/octree/src/octree_inst.cpp:37:
/opt/libpcl/pcl/octree/include/pcl/octree/octree_container.h: In member function ‘int pcl::octree::OctreeContainerEmpty::getPointIndex() const’:
/opt/libpcl/pcl/octree/include/pcl/octree/octree_container.h:152:50: error: ‘assert’ was not declared in this scope
     assert("getPointIndex: undefined point index");
                                                  ^
In file included from /opt/libpcl/pcl/octree/include/pcl/octree/octree_iterator.h:46:0,
                 from /opt/libpcl/pcl/octree/include/pcl/octree/octree2buf_base.h:44,
                 from /opt/libpcl/pcl/octree/include/pcl/octree/octree.h:41,
                 from /opt/libpcl/pcl/octree/src/octree_inst.cpp:37:
/opt/libpcl/pcl/octree/include/pcl/octree/octree_key.h: In copy constructor ‘pcl::octree::OctreeKey::OctreeKey(const pcl::octree::OctreeKey&)’:
/opt/libpcl/pcl/octree/include/pcl/octree/octree_key.h:60:78: error: ‘memcpy’ was not declared in this scope
   OctreeKey(const OctreeKey& source) { memcpy(key_, source.key_, sizeof(key_)); }
                                                                              ^
```

I'm having these CI errors also locally after the rest of the changes, I only realized now after finishing and testing the build for the actual new changes.

**Comment by aPonza** [comment]
As long as the rest is fine, should I squash some commits before resolving conflicts?

File: common/include/pcl/make_shared.h [hunk 4]
```diff
@@ -32,55 +32,15 @@
...
- * boost::allocate_shared or boost::make_shared will be invoked in case T has or
- * doesn't have a custom allocator, respectively.
+ * \file pcl/make_shared.h
  *
- * \see pcl::has_custom_allocator, PCL_MAKE_ALIGNED_OPERATOR_NEW
- * \tparam T Type of the object to create a pcl::shared_ptr of
- * \tparam Args Types for the arguments to pcl::make_shared
- * \param args List of arguments with which an instance of T will be constructed
- * \return pcl::shared_ptr of an instance of type T
+ * \warning Do not use this header, its contents have been moved to pcl/memory.h
```
Comment (taketwo) [inline common/include/pcl/make_shared.h]: I meant `#warning` actually :)

**Comment by aPonza** [comment]
What do both thumbs up on the previous comment mean? Squash into one commit?

**Comment by taketwo** [comment]
Sorry for the lack of clarity. I meant to support the idea of squashing the commits. From my side, a single commit would be good, but if you want to be more granular, that's also fine.

**Comment by kunaltyagi** [comment]
Same. Squash and rebase as appropriate, to make it easier to resolve conflicts

**Comment by aPonza** [comment]
What's happened to the Formatting and Documentation jobs? I'll rebase for the changes, so CI can run again, possibly fixing these two.

**Comment by aPonza** [comment]
Ready for merge, unless you first want to fix 19.10

**Comment by SergioRAgostinho** [comment]
@kunaltyagi shouldn't the label be "needs: review"?

**Comment by kunaltyagi** [comment]
Sorry about that. I hope to get fewer errors as days go by

**Comment by kunaltyagi** [comment]
@aPonza Could you resolve the conflicts please?

**Comment by aPonza** [comment]
This is https://github.com/PointCloudLibrary/pcl/issues/2601 and https://github.com/PointCloudLibrary/pcl/issues/3722

**Comment by SergioRAgostinho** [comment]
@PointCloudLibrary/maintainers Since we've decide to cherry pick for 1.10.1 can we merge this now?

**Pull request merged**